src/libs/pois.js

```diff
@@ -9,7 +9,19 @@ export function toUrl(poi) {
   if (poi.id === 'geolocalisation' || poi.type === 'latlon') {
     return `latlon:${poi.latLon.lat.toFixed(5)}:${poi.latLon.lng.toFixed(5)}`;
   }
-  return `${poi.id}@${ExtendedString.slug(poi.name)}`;
+  return poi.name ? `${poi.id}@${ExtendedString.slug(poi.name)}` : poi.id;
+}
+
+export async function getInputValue(poi) {
+  if (poi.type === 'latlon' || !poi.name) {
+    return await reverseGeocode(poi);
+  }
+  return poi.getInputValue();
+}
+
+async function reverseGeocode(poi) {
```

Comment:
`reverseGeocode` may be a confusing name for this function. In many cases, an address exists in the POI from the database, and no actual "reverse" is required.
Maybe `fetchAddressLabel` would be a better name ?

Reply:
You're right, I'll use your proposition.